csrc/scheduler/utils.cpp

```diff
@@ -2490,6 +2492,31 @@ std::unordered_set<TensorView*> getAllTvsFrom(
   return tv_group;
 }
 
+int64_t getSharedMemoryOverheadPerBlock(
+    Fusion* fusion,
+    const std::vector<TensorView*>& reduction_tvs) {
+  const auto& dev_prop = at::cuda::getCurrentDeviceProperties();
+  const int64_t max_threads_per_block = dev_prop->maxThreadsPerBlock;
```

Comment:
we are always using max threads per block here for computing?

Reply:
No. when this function is being called to do runtime check, we don't know how many threads will be used in the heuristics. So the max is used to be safe. I can put `max_threads_per_block` as a para, if the caller knows the value it can pass the exact value, otherwise, we will use the hardware max.